### ISSUE TITLE
Nextcloud 30 support

### ISSUE THREAD

**Issue by Glandos** [nextcloud/user_external#260]
**Is your feature request related to a problem? Please describe.**
I'm about to migrate to Nextcloud 30 (aka Hub 9), and wanted to know if this application is ready for it.

**Describe the solution you'd like**
Declare compatibility with NC 30.

**Describe alternatives you've considered**
Not migrating.

**Additional context**
I'm ready to test if needed.

See #240 and #246 for the usual patches.

**Comment by prismopensource** [comment]
Happy to sponsor a bit of time for @violoncelloCH if that's what's needed to get a Nextcloud 30 compatible release out of the door this fall.

**Comment by Glandos** [comment]
My first test with IMAP worked fine.

**Comment by ervee** [comment]
I just enabled the App in my NC30 test installation and configured it with IMAP. Works fine with my Dovecot backend. Also logged in from an Android Nexcloud App and that also works fine.

**Comment by ervee** [comment]
I updated my NextCloud v29 to v30 via ```sudo -E -u www-data /usr/bin/php --define apc.enable_cli=1 /var/www/nextcloud/updater/updater.phar``` and after the update was done enabled the user_external App via ```sudo -E -u www-data /usr/bin/php --define apc.enable_cli=1 occ app:install user_external --force```. This works fine for a couple of days now.

**Comment by ickedude** [comment]
Works fine with v30.0.2 and ssh backend.

**Comment by ervee** [comment]
I just updated to 30.0.2 and the user_external App got disabled. You actually see it in the occ upgrade output:
```
[...snip...]
Updating database schema
Updated database
Disabled incompatible app: user_external
Updating <circles> ...
Repair step: Upgrading Circles App
[...snip...]
```
Not sure if you would see this using other update methods.

After a quick ```occ app:install user_external --force``` (or force enable the App in the Admin UI) I can login in NC 30.0.2 without issue using IMAP.

**Comment by MCF** [comment]
Just updated to 30.0.2  The update screen in 29.0.8 showed user_external as **not** being compatible with version 30.0.2 - but it all just worked as soon as the update was completed.  There was no need to force install user_external.  Everything was fine

I'm using the SSH backend FWIW.

**Comment by jensb** [comment]
Yes, please enable this plugin for NC30! I also need it badly.

**Comment by alexhass** [comment]
Need the update, too.

**Comment by ervee** [comment]
I'm in the progress of migrating my users to LDAP, which is a pain in the butt. But I suggest to everyone to move away from this plugin.
There just is no development here and pull requests are not merged. And nobody is stepping up to fork and publish to the NextCloud App store.

**Comment by tflidd** [comment]
While you are at it, it would be great to add NC31 as well...

**Comment by prismopensource** [comment]
> While you are at it, it would be great to add NC31 as well...

[Nextcloud 31 support #264](https://github.com/nextcloud/user_external/issues/264)

**Comment by alexhass** [comment]
Where is the release?

**Comment by alexhass** [comment]
https://apps.nextcloud.com/apps/user_external shows no v30/31. whats wtong here? How should i install the addon?

**Comment by tflidd** [comment]
It needs to be released by someone with permissions. @violoncelloCH seems to be the official maintainer, not sure if someone from Nextcloud can help out.

In the meantime, you can just try to pull it directly from github, then the app is probably unsigned.

**Comment by ervee** [comment]
Yes that or just force install with ```occ app:install user_external --force```.

**Comment by Chrisi-Bavaria** [comment]
In my option it's really not a great idea to install a non supported plugin at this security critical task of authenticating users.
So please Nextcloud devs, test and support this app if the primary maintainer is not able to update for over a year...
It's really a pitty!

**Comment by alexhass** [comment]
There is no relase for v30. Why has the cased closed?

**Comment by AndyScherzinger** [comment]
Pre-release for 31+ is out!

Feedback welcome on https://github.com/nextcloud/user_external/issues/292
